frontend/public/components/catalog/catalog-items.jsx

```diff
@@ -66,79 +67,103 @@ export const catalogCategories = {
   },
 };
 
-// Filter property white list
-const filterGroups = [
-  'kind',
-];
-
-const getAvailableFilters = initialFilters => {
-  const filters = _.cloneDeep(initialFilters);
-  filters.kind = {
-    ClusterServiceClass: {
-      label: 'Service Class',
-      value: 'ClusterServiceClass',
-      active: false,
-    },
-    ImageStream: {
-      label: 'Source-to-Image',
-      value: 'ImageStream',
-      active: false,
-    },
-  };
-
-  return filters;
-};
-
-
-const filterGroupNameMap = {
-  kind: 'Type',
-};
-
-const filterValueMap = {
-  ClusterServiceClass: 'Service Class',
-  ImageStream: 'Source-to-Image',
-};
-
-const keywordCompare = (filterString, item) => {
-  if (!filterString) {
-    return true;
-  }
-  if (!item) {
-    return false;
-  }
-
-  return item.tileName.toLowerCase().includes(filterString) ||
-    (item.tileDescription && item.tileDescription.toLowerCase().includes(filterString)) ||
-    (item.tags && item.tags.includes(filterString));
-};
-
 const setURLParams = params => {
   const url = new URL(window.location);
   const searchParams = `?${params.toString()}${url.hash}`;
 
   history.replace(`${url.pathname}${searchParams}`);
 };
 
-export class CatalogTileViewPage extends React.Component {
+export class CatalogTileViewPage extends TileViewPage {
```

Comment:
I'm concerned with using inheritance here. It's considered an antipattern in react and has caused us some pain in other places (particularly when we need to connect components to Redux). Is there a way we can use composition instead?

https://reactjs.org/docs/composition-vs-inheritance.html

Reply:
I'll take a shot at it.